plugins/hashsum_download/server/__init__.py

```diff
@@ -118,7 +148,50 @@ def _getFirstFileByHash(self, algo, hash, user=None):
         return None
 
 
+def _computeHashHook(event):
+    """
+    Computes all supported checksums on a given file. Downloads the
+    file data and stream-computes all required hashes on it, saving
+    the results in the file document.
+    """
+    if ModelImporter.model('setting').get(PluginSettings.AUTO_COMPUTE, default=False):
+        _computeHash(event.info['file'])
+
+
+def _computeHash(file, progress=noProgress):
+    toCompute = SUPPORTED_ALGORITHMS - set(file)
+    toCompute = {alg: getattr(hashlib, alg)() for alg in toCompute}
+
+    if not toCompute:
+        return
+
+    fileModel = ModelImporter.model('file')
+    with fileModel.open(file) as fh:
+        while True:
+            chunk = fh.read(65536)
```

Comment:
You should probably move this to the top as a module level constant.

Maybe we should centralize this value, as we have it in a few places.  Is it well determined, or semi-arbitrary, both?

As [BUF_LEN](https://github.com/girder/girder/blob/70902befc45b165b6d5e9a34828c15ad6281814e/girder/utility/s3_assetstore_adapter.py#L33), [BUF_SIZE](https://github.com/girder/girder/blob/70902befc45b165b6d5e9a34828c15ad6281814e/girder/utility/filesystem_assetstore_adapter.py#L36), and [MAX_CHUNK_SIZE](https://github.com/girder/girder/blob/70902befc45b165b6d5e9a34828c15ad6281814e/clients/python/girder_client/__init__.py#L158).  Actually, it's probably better to keep them independent, and this isn't terribly important anyway.